### PR TITLE
fix: close dialogs when clicking outside

### LINKED ISSUE #985: Bug: close update dialog disables mouse event on page body
### Describe the bug

https://user-images.githubusercontent.com/62995161/224093042-a718091e-2260-4133-9d35-352a86fc1328.mov

### Steps to reproduce

1. Connect account 
2. Go to user profile page
3. Click on more icon
4. select `Edit`
5. click on the `X` close icon

### Affected services

insights.opensauced.pizza

### Platforms

_No response_

### Browsers

_No response_

### Environment

Production, Development

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by TAKANOME-DEV**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This pr fixes the issues with dialogs that won't close after clicking outside of the component.
- If the dialog has to be controlled, two props must be passed to the radix portal so that it will close the dialog if the outside is clicked. Those props are `open` and `onPageChange`
- Note that for deleting a highlight, we're using [radix alert](https://www.radix-ui.com/docs/primitives/components/alert-dialog) which won't close when clicking outside.
- Unrelated but the dialog showing a single highlight is too small on my end.
## Related Tickets & Documents

Fixes #985 
## Mobile & Desktop Screenshots/Recordings

https://www.loom.com/share/32833c5e8fd44d73b90cf9c607461304
## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by brandonroberts** [comment]
@TAKANOME-DEV this works if I click a link on the page to open the modal, but now it doesn't open the modal when navigating directly. 

https://user-images.githubusercontent.com/42211/225027058-51fc9bfc-bc98-4dcb-b358-0039e3c7d1ea.mov

**Comment by TAKANOME-DEV** [comment]
There are some hydration issues on dev, but I can't localize the elements causing it
<details>
<summary>Error Stack</summary>

```sh
# -------------------------------------------------------------
Warning: Expected server HTML to contain a matching <div> in <div>.
# -------------------------------------------------------------
div
div
$8927f6f2acc4f386$export$250ffa63cdc0d034</Node<@webpack-internal:///./node_modules/@radix-ui/react-tabs/node_modules/@radix-ui/react-primitive/dist/index.module.js:44:58
$921a889cee6df7e8$export$99c2b779aa4e8b8b@webpack-internal:///./node_modules/@radix-ui/react-presence/dist/index.module.js:30:56
$69cb30bb0017df05$export$bd905d70e8fd2ebb<@webpack-internal:///./node_modules/@radix-ui/react-tabs/dist/index.module.js:157:121
# -------------------------------------------------------------
_c4@webpack-internal:///./components/atoms/Tabs/tabs.tsx:48:36
# -------------------------------------------------------------
div
$8927f6f2acc4f386$export$250ffa63cdc0d034</Node<@webpack-internal:///./node_modules/@radix-ui/react-tabs/node_modules/@radix-ui/react-primitive/dist/index.module.js:44:58
Provider@webpack-internal:///./node_modules/@radix-ui/react-context/dist/index.module.js:48:72
$69cb30bb0017df05$export$b2539bed5023c21c<@webpack-internal:///./node_modules/@radix-ui/react-tabs/dist/index.module.js:55:236
div
# -------------------------------------------------------------
Feeds@webpack-internal:///./pages/feed/index.tsx:43:92
# -------------------------------------------------------------
main
div
# -------------------------------------------------------------
ProfileLayout@webpack-internal:///./layouts/profile.tsx:13:25
# -------------------------------------------------------------
Provider@webpack-internal:///./node_modules/@radix-ui/react-context/dist/index.module.js:48:72
$a093c7e1ec25a057$export$f78649fb9ca566b8@webpack-internal:///./node_modules/@radix-ui/react-tooltip/dist/index.module.js:73:252
SessionContextProvider@webpack-internal:///./node_modules/@supabase/auth-helpers-react/dist/index.js:41:30
SWRConfig@webpack-internal:///./node_modules/swr/_internal/dist/index.mjs:557:24
MyApp@webpack-internal:///./pages/_app.tsx:42:38
PathnameContextProviderAdapter@webpack-internal:///./node_modules/next/dist/shared/lib/router/adapters.js:62:34
ErrorBoundary@webpack-internal:///./node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:301:63
ReactDevOverlay@webpack-internal:///./node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:850:908
Container@webpack-internal:///./node_modules/next/dist/client/index.js:62:1
AppContainer@webpack-internal:///./node_modules/next/dist/client/index.js:172:25
Root@webpack-internal:///./node_modules/next/dist/client/index.js:347:37
```

</details

There is also this issue probably caused by `router.push` -> `Error: Abort fetching component for route: "/feed"`. Probably only visible on dev, but I don't know how to fix it yet https://stackoverflow.com/a/75702362/15301961

File: pages/feed/index.tsx [hunk 1]
```diff
@@ -41,20 +43,26 @@ const Feeds = () => {
 
   useEffect(() => {
     if (selectedRepo) {
-      router.push(`/feed?repo=${selectedRepo}`);
+      router.push(`/feed?repo=${selectedRepo}`).catch(console.error);
```
Comment (0-vortex) [inline pages/feed/index.tsx:46]: IMHO this is not caught because it is not awaited

**Review by 0-vortex — CHANGES_REQUESTED** [review]
You need to update your branch with beta to trigger the latest release code and return or await the router redirects you implemented in the feed in order to be able to debug them correctly. 

It's likely this is 95% ready, just need to get rid of that error 👍

File: pages/feed/index.tsx (hunk 1 above)
Comment (TAKANOME-DEV) [reply pages/feed/index.tsx]: Yeah :sweat_smile: 
> calling router.push() multiple times within the same Next.js page in different places/throughout different re-renders seems to cause this error in some cases
source: https://stackoverflow.com/a/73344411/15301961

**Comment by brandonroberts** [comment]
@TAKANOME-DEV still seeing the same issue with the modal closing on page load

**Comment by TAKANOME-DEV** [comment]
> @TAKANOME-DEV still seeing the same issue with the modal closing on page load
>

Yeah, honestly I don't know how to fix it yet, the component re-renders multiple times that’s why it closes. I will investigate more on it, any help is appreciated.

File: pages/feed/index.tsx [hunk 2]
```diff
@@ -41,22 +44,35 @@ const Feeds = () => {
...
 
+
   useEffect(() => {
     if (selectedRepo) {
       router.push(`/feed?repo=${selectedRepo}`);
     }
+
     if (highlightId) {
-      router.push(`/feed/${id}`);
+      setOpenSingleHighlight(true);
```
Comment (OgDev-01) [inline pages/feed/index.tsx:54]: Hey @TAKANOME-DEV , you might need to add 
```javascript
router.push(`/feed/${id}`);
```
below the `setOpenSingleHighlight(true)` to persist the `{id}` on page reload.

Comment (TAKANOME-DEV) [reply pages/feed/index.tsx:54]: Yeah, but it causes errors in the console. The issue was caused by this:
```js
if (!highlightId && !selectedRepo) {
  router.push("/feed");
}
```

**Comment by TAKANOME-DEV** [comment]
@brandonroberts it should be good now

**Comment by OgDev-01** [comment]
The new fix introduces another bug 

https://user-images.githubusercontent.com/62995161/226817777-40717b9f-8704-4e8c-b5c3-baf89f46f5f4.mov

viewing a single highlight is now blocked when a filter is selected selected

**Comment by OgDev-01** [comment]
Works as expected now 🍕

**Comment by TAKANOME-DEV** [comment]
> Works as expected now pizza
>
Not yet :sweat_smile: When removing the filter, it still stays in the URL I think
:sweat_smile:

File: components/molecules/ContributorHighlight/contributor-highlight-card.tsx [hunk 3]
```diff
@@ -151,231 +151,224 @@ const ContributorHighlightCard = ({ title, desc, prLink, user, id }: Contributor
...
+                  </a>
+                </DropdownMenuItem>
+              </DropdownMenuContent>
+            </DropdownMenu>
           </div>
-          {/* Highlight Link section */}
+        </div>
 
-          <div>
-            <a href={prLink} className="underline text-sauced-orange cursor-pointer">
```
Comment (OgDev-01) [inline components/molecules/ContributorHighlight/contributor-highlight-card.tsx:259]: @bdougie, should this be changed to a regular `p` element or given the `target="_blank"` attribute to open the link in a new tab?

Comment (brandonroberts) [reply components/molecules/ContributorHighlight/contributor-highlight-card.tsx:259]: Adding `target=_blank` would be good

**Review by brandonroberts — APPROVED** [review]
Works as expected

**Review by OgDev-01 — APPROVED** [review]

**Comment by brandonroberts** [comment]
@TAKANOME-DEV will you resolve the merge conflicts, then we're ready to merge

**Pull request merged**